pkg/devfile/parser/parse.go

```diff
@@ -392,10 +397,38 @@ func parseFromURI(importReference v1.ImportReference, curDevfileCtx devfileCtx.D
 	if !absoluteURL && curDevfileCtx.GetAbsPath() != "" {
 		newUri = path.Join(path.Dir(curDevfileCtx.GetAbsPath()), uri)
 		d.Ctx = devfileCtx.NewDevfileCtx(newUri)
+		if !strings.HasSuffix(newUri, ".yaml") {
```

Comment:
I'm wondering if it's reasonable to check if it's a `.yaml` format file. we've never discussed this in detail before. what if the devfile path provided contains yaml content, but it does not end with `.yaml`?

Reply:
I remember an issue got opened several months ago saying we should support `.yml` extensions too https://github.com/devfile/api/issues/744

Reply:
I removed this check as it’s not needed here

I think the file format is being checked in the context here: https://github.com/devfile/library/blob/main/pkg/devfile/parser/context/context.go#L97-L98